.rubocop.yml

```diff
@@ -1,63 +1,79 @@
+# List of cops can be found here:
+# https://github.com/bbatsov/rubocop/blob/master/config/default.yml
+# https://github.com/bbatsov/rubocop/blob/master/config/disabled.yml
+# https://github.com/bbatsov/rubocop/blob/master/config/enabled.yml
+require: rubocop-rspec
+
 inherit_from: .rubocop_todo.yml
 
 AllCops:
   DisplayCopNames: true
-  Include:
-    - '**/Rakefile'
-    - '**/config.ru'
+  DisplayStyleGuide: true
   Exclude:
-    - 'db/**/*'
-    - 'script/**/*'
-    - 'spec/internal/**/*'
-    - 'spec/test_app_templates/**/*'
     - 'vendor/**/*'
+    - 'db/**/*'
+    - 'tmp/**/*'
+  ExtraDetails: true
+  TargetRubyVersion: 2.2
 
-Lint/ImplicitStringConcatenation:
-  Exclude:
-    - 'lib/generators/sufia/**/*'
-
-Metrics/LineLength:
-  Max: 400
+# Alot of these metrics are quite restrictive. We may want to disable a few of these.
+# For now let's keep them and increase them by an excessive amount
 
+# A calculated magnitude based on number of assignments,
+# branches, and conditions.
 Metrics/AbcSize:
-  Max: 30
-  Exclude:
-    - 'app/controllers/concerns/sufia/admin/depositor_stats.rb'
-    - 'app/controllers/concerns/sufia/my_controller_behavior.rb'
-    - 'app/services/sufia/user_stat_importer.rb'
-    - 'lib/sufia/arkivo/metadata_munger.rb'
+  Max: 45 # default 15
 
+Metrics/ClassLength:
+  Max: 500 # default 100
+
+# A complexity metric that is strongly correlated to the number
+# of test cases needed to validate a method.
+Metrics/CyclomaticComplexity:
+  Max: 17 # default 6
+
+Metrics/LineLength:
+  Enabled: true
+  Max: 120 # default is 80
+
+# Avoid methods longer than 10 lines of code.
 Metrics/MethodLength:
-  Max: 15
+  Max: 40 # default 10
 
-Style/SymbolProc: # https://github.com/bbatsov/rubocop/issues/3071
-  Exclude:
-    - 'app/models/sufia/uploaded_file.rb'
+Metrics/ModuleLength:
+  Max: 500 # default 100
 
-Style/IndentationConsistency:
-  EnforcedStyle: rails
+# A complexity metric geared towards measuring complexity for a human reader.
+Metrics/PerceivedComplexity:
+  Max: 18 # default 7
```

Comment:
Same comment as the cyclometric complexity I guess. I'd rather have someone say "hey, this is too complex/dense for me to read" than have a bunch of heuristics half-ass the sentiment.